app/models/post.rb

```diff
@@ -652,8 +663,30 @@ def hide!(post_action_type_id, reason = nil, custom_message: nil)
   def unhide!
     Post.transaction do
       self.update!(hidden: false)
-      self.topic.update(visible: true) if is_first_post?
-      UserStatCountUpdater.increment!(self)
+      should_update_user_stat = true
+
+      # NOTE: We have to consider `nil` a valid reason here because historically
+      # topics didn't have a visibility_reason_id, if we didn't do this we would
+      # break backwards compat since we cannot backfill data.
+      if is_first_post? &&
+           (
+             self.topic.visibility_reason_id ==
+               Topic.visibility_reasons[:op_flag_threshold_reached] ||
+               self.topic.visibility_reason_id.nil?
```

Comment:
Wondering if this condition could be named for increased clarity. WDYT? 🤔

Reply:
Yeah good point, will think of something